### PR TITLE
178 Refactor `adab` to add new `PARAM`s 

### LINKED ISSUE #178: Refactor `adab` with more `PARAM` values
Add `PARAM`s to `adab` including those needed for ADAT04A:

- [x] `ADASTAT1` (ADA Status of a patient)
- [x] `EMERNEG1` (Treatment emergent negative ADA)
- [x] `EMERPOS1` (Treatment emergent positive ADA)
- [x] `TRUNAFF1` (ADA treatment unaffected)
- [x] `BFLAG1` (Baseline) 
- [x] `EMERNEG2` (Treatment emergent negative NAb)
- [x] `EMERPOS2` (Treatment emergent positive NAb)
- [x] `TRUNAFF2` (NAb treatment unaffected)
- [x] `BFLAG2` (Baseline NAb) 
- [x] `INDUCD2` (Treatment-induced NAb)
- [x] `ENHANC2` (Treatment-enhanced NAb)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by edelarua**
Closes #178

### CODE REVIEW

File: R/radab.R
Comment (codesformochi) [inline R/radab.R:31]:
```suggestion
                    "ADA Treatment unaffected", "Treatment Emergent - Negative ADA",
```

File: R/radab.R
Comment (codesformochi) [inline R/radab.R:32]:
```suggestion
                    "Treatment Emergent - Positive ADA", "Persistent ADA", "Transient ADA",
```

File: R/radab.R
Comment (codesformochi) [inline R/radab.R:38]:
```suggestion
                    "Treatment unaffected ADA, Neutralizing Antibody"
```

File: R/radab.R
Comment (codesformochi) [inline R/radab.R:215]:
```suggestion
        (PARAM == "Treatment unaffected ADA" & pos_bl & (inc_postbl == 0 | !any_pos_postbl)) ~ "Y",
```

Comment (codesformochi) [reply R/radab.R:215]: suggest similar changes below

**Review by codesformochi — COMMENTED** [review]
looks good and runs! suggested small label changes.

**Comment by edelarua** [comment]
Hi @codesformochi, thanks for the review! I have updated the `PARAM` labels according to your suggestions. 

Please also note that there are updates to multiple cached datasets since `radsl` was recently refactored causing a change in the cached datasets which take the adsl data as an argument.

**Comment by edelarua** [comment]
@khatril Lyndsee has reviewed this PR and suggested minor label changes which I implemented. Do you want to do a final review to merge this so that your ADAT04A template issue is unblocked?

**Comment by shajoezhu** [comment]
hi @edelarua , Leena is not available this afternoon. 

Hi @qiliu1013 @legrasv @stevedgthb and @mingyu528 , the label change is in commit https://github.com/insightsengineering/random.cdisc.data/pull/179/commits/3b0f73aa9b0ab0fe07fd4399433e85ea49fccc62

can you take a look please, should we use "ADA"? or "NAbs"?

**Comment by qiliu1013** [comment]
> hi @edelarua , Leena is not available this afternoon.
> 
> Hi @qiliu1013 @legrasv @stevedgthb and @mingyu528 , the label change is in commit [3b0f73a](https://github.com/insightsengineering/random.cdisc.data/commit/3b0f73aa9b0ab0fe07fd4399433e85ea49fccc62)
> 
> can you take a look please, should we use "ADA"? or "NAbs"?

Hi @shajoezhu, the previous version looks good. No need to add ADA or NAbs into the PARAMs. If the PARAMs relates NAbs, it can be tell from the "Neutralizing Antibody" in the PARAMs.

Thanks,
Qi

**Comment by edelarua** [comment]
> > hi @edelarua , Leena is not available this afternoon.
> > Hi @qiliu1013 @legrasv @stevedgthb and @mingyu528 , the label change is in commit [3b0f73a](https://github.com/insightsengineering/random.cdisc.data/commit/3b0f73aa9b0ab0fe07fd4399433e85ea49fccc62)
> > can you take a look please, should we use "ADA"? or "NAbs"?
> 
> Hi @shajoezhu, the previous version looks good. No need to add ADA or NAbs into the PARAMs. If the PARAMs relates NAbs, it can be tell from the "Neutralizing Antibody" in the PARAMs.
> 
> Thanks, Qi

Thanks @qiliu1013, I will revert the label change.

**Comment by edelarua** [comment]
@qiliu1013 Lyndsee has reviewed this PR for all added `PARAM`s except `BFLAG1` and `BFLAG2`. If you have time today couldyou provide a final review on the latest commit so we can unblock the ADAT04 templates?

**Review by qiliu1013 — APPROVED** [review]
Looks good.

**Pull request merged**
